### PR TITLE
test(prover-ray): compilers in depth testing

### LINKED ISSUE #2915: Write a design document describing the prover-ray compilation suite
## Summary

Produce a design document that describes the full compilation pipeline in `prover-ray`, from raw arithmetization constraints to the final Vortex proof.

## Motivation

The pipeline spans multiple non-trivial stages (log-derivative reduction, global/local constraint compilation, MPTS, Vortex commitment, self-recursion). A single reference document will align contributors and serve as an entry point for auditors.

## Suggested structure

1. Overview of the pipeline stages (memory-bus → lookup → log-derivative → global constraints → local constraints/openings → split-extensions → MPTS → Vortex + self-recursion)
2. Data flow between stages
3. Key design decisions and trade-offs (e.g. dynamic sizing, univariate vs multilinear mode)
4. Open questions / items pending decision

## Placement

Following the documentation philosophy of the project, this should live as a `README.md` or dedicated `.md` file within the `doc` folder.

### LINKED ISSUE #2921: Bootstrap heavy-testing framework to ensure max coverage in the cryptographic function path.
## Summary

Add a systematic soundness test harness for `prover-ray` that exercises every constraint and compiler pass with invalid witnesses, ensuring that every line of proving code is tested in every direction.

Specifically, we want to push up:
- [] The testing coverage
- [] The correctness
- [] Negative cases fuzzing 
- [] Race condition detection

## Motivation

A circuit or compiler pass tested only with valid witnesses is untested against its core security property. This harness should give us high confidence that unsatisfied constraints reliably cause proof generation to fail — not just that satisfied constraints produce valid proofs.

## Scope

- For every constraint type (global, local, log-derivative, RangeCheck, …), add at least one test that provides a deliberately invalid witness and asserts that proof generation fails
- Consider a fuzzing layer (`FuzzFoo` with `fuzzlight` build tag) for the witness assignment path to catch edge cases at boundaries. But the fuzzing should not go in CI.
- Deterministic seeds (e.g. `chacha8rand`) so failures are always reproducible
- Tests should run within a 2 minute window and be easy to run in parallel

### PR DESCRIPTION
**Pull request by AlexandreBelling**
This PR closes the following issues

Closes #2915 
Closes #2921 

Namely, it

* Documents the compilation pipeline
* Heavily unit-test all the compilation steps
* Adds general full-pipeline tests
* Remove support for the permutation columns. Was due in the past PR.

### Checklist

* [ ] I wrote new tests for my new core changes.
* [ ] I have successfully ran tests, style checker and build against my new changes locally.
* [ ] If this change is deployed to any environment (including Devnet), E2E test coverage exists or is included in this
  PR.
* [ ] I have informed the team of any breaking changes if there are any.

---

> [!NOTE]
> **Medium Risk**
> Breaking WIOP API (permutation removed) and non-trivial FFT/global prover changes could affect proof correctness; risk is mitigated by extensive new pipeline and soundness tests.
> 
> **Overview**
> This PR **deepens compiler coverage** and **documents** the WIOP/Arcane lowering story, while **narrowing** table relations to lookup/inclusion only.
> 
> **Documentation:** Adds `docs/arcane-compilation-suite.md` (pipeline from lookups/range/message-bus → log-derivative → local/global PLONK quotients). Removes the draft `ideas_for_dynamic_vortex.md`. Refreshes `wiop/class_diagram.mmd` to match the current API.
> 
> **API / queries:** Replaces `TableRelation` + `TableRelationKind` with **`LookupQuery`** (`NewInclusion` only). **Permutation** queries, `NewPermutation`, and related tests are **removed**. `System.TableRelations` is now `[]*LookupQuery`.
> 
> **Global compiler:** Quotient prover can use **extension-field witness columns** on the large coset (`reevalOnLargeCosetExt`, split `cosetEvals` / `cosetEvalsExt`). Adds **bit-reversal** steps around small/large FFT and ratio &gt; 1 quotient chunking so coset→quotient splitting matches DIF/DIT conventions.
> 
> **Lookup compiler:** When inclusion only touches **precomputed** B columns, `witnessRound` no longer points at `PrecomputedRound` (avoids runtime invariant breaks); defaults to the first interactive round and backfills groups.
> 
> **Testing:** New `wiop/compilers` package with **`compileFullPipeline`** end-to-end tests (range → lookup → LDS → local vanishing → global). Large shared **`wioptest`** scenario sets for local vanishing, LDS, lookup (honest + soundness), range check; per-pass `*_WioptestCompleteness/Soundness` tests including tampered **M**, **Z**, and **LDS result** paths.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 1861f5ab49d4e5c5698088f12b99a531664f627e. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: prover-ray/docs/compiler-architecture.png
Comment (Soleimani193) [inline prover-ray/docs/compiler-architecture.png:1]: We don't see the contribution of logDeriveSum in the compilation diagram.

File: prover-ray/wiop/compilers/lookuptologderivsum/context.go [hunk 1]
```diff
@@ -15,7 +15,7 @@ type includedSpec struct {
 	// query is the source [wiop.TableRelation] this fragment came from. The
```
Comment (Soleimani193) [inline prover-ray/wiop/compilers/lookuptologderivsum/context.go:15]: make sure to remove TableRelation from the comments

**Review by ThomasPiellard — DISMISSED** [review]

**Review by Soleimani193 — DISMISSED** [review]

File: prover-ray/docs/compiler-architecture.png
Comment (AlexandreBelling) [reply prover-ray/docs/compiler-architecture.png:1]: It was abbreviated to logup as log-derivative sum is small

Comment (AlexandreBelling) [reply prover-ray/docs/compiler-architecture.png:1]: Oh, I see... You right. I'll use direct draw.io embedding instead so that we can edit

**Review by Soleimani193 — APPROVED** [review]

**Review by ThomasPiellard — APPROVED** [review]
